HopsanGUI/Widgets/TextEditorWidget.cpp

```diff
@@ -589,7 +609,59 @@ void TextEditor::wheelEvent(QWheelEvent* event)
 //! @brief Updates list of auto complete words
 void TextEditor::updateAutoCompleteList()
 {
-    mpCompleter->setModel(new QStringListModel(gpTerminalWidget->mpHandler->getAutoCompleteWords(), mpCompleter));
+    if(mLanguage == HighlighterTypeEnum::Hcom) {
+        mpCompleter->setModel(new QStringListModel(gpTerminalWidget->mpHandler->getAutoCompleteWords(), mpCompleter));
+    }
+    else if(mLanguage == HighlighterTypeEnum::Cpp) {
+        QStringList lines = toPlainText().split("\n");
+
+        QStringList dataTypes = QStringList() << "size_t" << "double" << "int" << "SecondOrderTransferFunction" << "FirstOrderTransferFunction" << "Port";
+        QStringList functions = QStringList() << "addInputVariable" << "addOutputVariable" << "addConstant" << "addPowerPort" << "getSafeNodeDataPtr";
+
+        int bracketCounter=-1;
+
+        QStringList variables;
+        foreach(const QString &line, lines)
+        {
+            if(line.simplified().startsWith("class "))
```

Comment:
This may fail (reset counter) for nested classes, but not such a big deal I guess

Reply:
True, but I doubt anyone will expect the Hopsan autocompleter to support nested classes anyway.